### ISSUE TITLE
is support slip-0048?

### ISSUE THREAD

**Issue by tradzero** [Bit-Wasp/bitcoin-php#774]
i try to use this amazon project to create EOS mnemonic and private key, but seems it cannot get the correct private key here. so is [slip-0048](https://github.com/satoshilabs/slips/blob/master/slip-0048.md) support? 
and another question can i create uncompressed wif private key? 

here is my code
```php
<?php

require_once 'vendor/autoload.php';

use BitWasp\Bitcoin\Crypto\Random\Random;
use BitWasp\Bitcoin\Key\Factory\HierarchicalKeyFactory;
use BitWasp\Bitcoin\Mnemonic\Bip39\Bip39Mnemonic;
use BitWasp\Bitcoin\Mnemonic\Bip39\Bip39SeedGenerator;
use BitWasp\Bitcoin\Mnemonic\MnemonicFactory;
use BitWasp\Bitcoin\Bitcoin;

// Generate a mnemonic
$random = new Random();
$entropy = $random->bytes(Bip39Mnemonic::MAX_ENTROPY_BYTE_LEN);
$bip39 = MnemonicFactory::bip39();
$seedGenerator = new Bip39SeedGenerator();
$mnemonic = $bip39->entropyToMnemonic($entropy);
// Derive a seed from mnemonic/password
$seed = $seedGenerator->getSeed($mnemonic, '');
$hdFactory = new HierarchicalKeyFactory();
$bip32 = $hdFactory->fromEntropy($seed);
$bip32 = $bip32->derivePath("48'/4'/0'/0'/0'");
$privateKey = $bip32->getPrivateKey()->toWif();
echo($privateKey);
```

**Comment by afk11** [comment]
Hi, thanks for including code & a link to the spec in your issue!

Scanning through it, it seems the spec only covers the structure of the BIP32 path (roles, cosigners, networks, etc). I would expect the library to produce the right result here, as the spec is just a BIP32 path structure, and we have BIP32 fairly well tested!

You mentioned it's deriving the wrong private key - but I don't see a test fixture to follow in the SLIP. Could you share the mnemonic you were testing with, and the key you expected to get? Where did this information come from?

**Comment by tradzero** [comment]
OK. this infomation is came from chinese eos community, actually i also cannot get the mnemonic that can support test, but i will try hard to find one. thanks you so much to help that.
By the way. i knew how to get the uncompressed private key.

**Comment by afk11** [comment]
you might be able to use an eos node, avoid downloading the chain, and generate a wallet to compare with that way?

**Comment by tradzero** [comment]
Thanks for your advice again. But EOS seems not officially support mnemonic. Someone in chinese eos community said `m / 44'/ 194'/ 0'/ 0/0` suit for single private key chain, and like Graphene, one account may have many private key, so that recommend use slip-0048. and i am asking them offer me a private key can test.

**Comment by tradzero** [comment]
I got a test pair. And i found it works. here is mnemonic `enact chase company loyal goddess noise focus change column weather tourist life tuna width range thunder border merit goose muffin umbrella cushion empty panda` and uncompressed private key is `5JGbhqTGAChGjG5yyx4BMgf7KcXCqwWQh5FTYwFQgGKgP3FLbhu`.
I found i cannot get the right private key because i misunderstanding the right path, and right path is `m/48'/4'/1'/0'/0'` it is for active role. and `48'/4'/0'/0'/0'` is for owner role.
Thanks for your amazing project again.
